CMakeLists.txt

```diff
@@ -314,6 +314,16 @@ if (LBANN_HAS_CUDA)
   else ()
     set(LBANN_HAS_NCCL2 FALSE)
   endif ()
+
+  if (LBANN_WITH_NVSHMEM)
+    find_package(NVSHMEM REQUIRED)
+    set_property(TARGET cuda::toolkit PROPERTY
+      INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CUDA>:-arch=sm_70>)
```

Comment:
Do you need this in addition to the `CMAKE_CUDA_FLAGS` setting in FindNVSHMEM?

Reply:
I put this to suppress compilation with the default architecture, which I believe is sm_30. This was also to override when Hydrogen manually sets the architecture to sm_30.